src/main/java/com/odysseusinc/arachne/portal/service/submission/impl/BaseSubmissionServiceImpl.java

```diff
@@ -549,27 +550,43 @@ public void uploadCompressedResultsByDataOwner(Long submissionId, File compresse
         IUser user = userService.getByUsername(userDetails.getUsername());
 
         Path unzipDir = Files.createTempDirectory(String.format("submission_%d_results", submissionId));
-        ZipFile zipFile = new ZipFile(compressedFile);
-        final List<FileHeader> fileHeaders = zipFile.getFileHeaders().stream()
-                .filter(fileHeader -> !fileHeader.isDirectory())
-                .collect(Collectors.toList());
 
         try {
-            for (FileHeader fileHeader : fileHeaders) {
-                String relativeFilePath = fileHeader.getFileName();
-                String relativePath = FilenameUtils.getPath(relativeFilePath);
-                if(isNotBlank(relativePath)){
-                    unzipDir.resolve(relativePath).toFile().mkdirs();
-                }
-                final File localFile = unzipDir.resolve(relativeFilePath).toFile();
-                FileUtils.copyInputStreamToFile(zipFile.getInputStream(fileHeader), localFile);
-            }
+            unzipRecursively(new ZipFile(compressedFile), unzipDir, MAX_NESTING_FOR_UNZIP_INNER_ZIP_FILES);
             storeExtractedFiles(submission, unzipDir, user.getId());
         } finally {
             FileUtils.deleteDirectory(unzipDir.toFile());
         }
     }
 
+    private void unzipRecursively(final ZipFile zipFile, final Path unzipDir, final int nesting) throws IOException {
+        final List<FileHeader> fileHeaders = zipFile.getFileHeaders().stream()
+                .filter(fileHeader -> !fileHeader.isDirectory())
+                .collect(Collectors.toList());
+        for (FileHeader fileHeader : fileHeaders) {
+            final String relativeFilePath = fileHeader.getFileName();
+            final String relativePath = FilenameUtils.getPath(relativeFilePath);
+            if(isNotBlank(relativePath)){
+                unzipDir.resolve(relativePath).toFile().mkdirs();
+            }
+            final File localFile = unzipDir.resolve(relativeFilePath).toFile();
+            FileUtils.copyInputStreamToFile(zipFile.getInputStream(fileHeader), localFile);
+
+            if (relativeFilePath.endsWith(".zip") && nesting > 0) {
```

Comment:
Why we should limit the level of archive nesting? 
Anyway, what is the reason to execute the procedure when the limit is exceeded?

Reply:
agreed. nesting limit removed